pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -0,0 +1,161 @@
+        gpu_utilization: bool = True,
+        intra_step_time: bool = False,
+        inter_step_time: bool = False,
+        fan_speed: bool = False,
+        temperature: bool = False
+    ):
+        super().__init__()
+
+        if shutil.which("nvidia-smi") is None:
+            raise MisconfigurationException(
+                'Cannot use GPUStatsMonitor callback because NVIDIA driver is not installed.'
+            )
+
+        self.memory_utilization = memory_utilization
+        self.gpu_utilization = gpu_utilization
+        self.intra_step_time = intra_step_time
+        self.inter_step_time = inter_step_time
+        self.fan_speed = fan_speed
+        self.temperature = temperature
+
+    def on_batch_start(self, trainer, pl_module):
+        if self.gpu_utilization:
+            self._log_gpu(trainer)
+        if self.memory_utilization:
+            self._log_memory(trainer)
+
+        if self.inter_step_time and self.snap_inter_step_time:
+            # First log at beginning of second step
+            trainer.logger.log_metrics(
+                {'batch_time/inter_step (ms)': (time.time() - self.snap_inter_step_time) * 1000},
+                step=trainer.global_step
+            )
+
+        if self.intra_step_time:
+            self.snap_intra_step_time = time.time()
+
+    def on_batch_end(self, trainer, pl_module):
+        if self.gpu_utilization:
+            self._log_gpu(trainer)
+
+        if self.memory_utilization:
+            self._log_memory(trainer)
+
+        if self.fan_speed:
+            trainer.logger.log_metrics(self._get_gpu_stat("fan.speed", "%"), step=trainer.global_step)
+
+        if self.temperature:
+            trainer.logger.log_metrics(self._get_gpu_stat("temperature.gpu", "degrees C"), step=trainer.global_step)
+            trainer.logger.log_metrics(self._get_gpu_stat("temperature.memory", "degrees C"), step=trainer.global_step)
+
+        if self.inter_step_time:
+            self.snap_inter_step_time = time.time()
+
+        if self.intra_step_time and self.snap_intra_step_time:
+            trainer.logger.log_metrics(
+                {'batch_time/intra_step (ms)': (time.time() - self.snap_intra_step_time) * 1000},
+                step=trainer.global_step
+            )
+
+    def on_train_start(self, trainer, pl_module):
+        if not trainer.logger:
+            raise MisconfigurationException(
+                'Cannot use GPUStatsMonitor callback with Trainer that has no logger.'
+            )
+
+        if not trainer.on_gpu:
+            rank_zero_warn(
+                'You are using GPUStatsMonitor but are not running on GPU.'
+                ' Logged utilization will be independent from your model.', RuntimeWarning
+            )
+
+    def on_train_epoch_start(self, trainer, pl_module):
+        self.snap_intra_step_time = None
+        self.snap_inter_step_time = None
+
+    @staticmethod
+    def _get_gpu_stat(pitem: str, unit: str):
+        result = subprocess.run(
```

Comment:
how about if NVIDIA is not instaled, will it pass or fail?

Reply:
there is a MisconfigurationException already added in `__init__`.